atomics/include/desul/atomics/Common.hpp

```diff
@@ -50,4 +52,74 @@ struct MemoryScopeDevice {};
 // Core scoped (i.e. a shared Level 1 cache)
 struct MemoryScopeCore {};
 }  // namespace desul
+
+#ifndef __ATOMIC_RELAXED
+#define __ATOMIC_RELAXED 0
+#define __ATOMIC_CONSUME 1
+#define __ATOMIC_ACQUIRE 2
+#define __ATOMIC_RELEASE 3
+#define __ATOMIC_ACQ_REL 4
+#define __ATOMIC_SEQ_CST 5
+#endif
+
+namespace desul {
+template <class MemoryOrderDesul>
+struct GCCMemoryOrder;
+
+template <>
+struct GCCMemoryOrder<MemoryOrderRelaxed> {
+  static constexpr int value = __ATOMIC_RELAXED;
+};
+
+template <>
+struct GCCMemoryOrder<MemoryOrderSeqCst> {
+  static constexpr int value = __ATOMIC_SEQ_CST;
+};
+template <>
+struct GCCMemoryOrder<MemoryOrderAcquire> {
+  static constexpr int value = __ATOMIC_ACQUIRE;
+};
+template <>
+struct GCCMemoryOrder<MemoryOrderRelease> {
+  static constexpr int value = __ATOMIC_RELEASE;
+};
+
+template <class MemoryOrderDesul>
+struct CXXMemoryOrder;
+
+template <>
+struct CXXMemoryOrder<MemoryOrderRelaxed> {
+  static constexpr std::memory_order value = std::memory_order_relaxed;
+};
+
+template <>
+struct CXXMemoryOrder<MemoryOrderSeqCst> {
+  static constexpr std::memory_order value = std::memory_order_seq_cst;
+};
+template <>
+struct CXXMemoryOrder<MemoryOrderAcquire> {
+  static constexpr std::memory_order value = std::memory_order_acquire;
+};
+template <>
+struct CXXMemoryOrder<MemoryOrderRelease> {
+  static constexpr std::memory_order value = std::memory_order_release;
+};
+}
+
+namespace desul {
+namespace Impl {
+  template<class T>
+  struct numeric_limits_max;
+
+  template<>
+  struct numeric_limits_max<uint32_t> {
+    static constexpr uint32_t value = 0xffffffffu;
+  };
+  template<>
+  struct numeric_limits_max<uint64_t> {
+    static constexpr uint64_t value = 0xfffffffflu;
+  };
+
+}
+}
```

Comment:
Can you add a comment explaining why this isn't just `std::numeric_limits`?

Reply:
Yeah. It didn't like it on CUDA without experimental const expr relaxed blah blah thingy.

Reply:
Oh right, because `max()` is a static member function and not a constant. Ugh...